### PR TITLE
Docs: Update HTTP API docs for nested folders

### PR DESCRIPTION
**Pull request by papagian**
**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes partially #65419

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/developers/http_api/folder.md [hunk 1]
```diff
@@ -35,6 +35,10 @@ that you cannot use this API for retrieving information about the General folder
 
 Returns all folders that the authenticated user has permission to view. You can control the maximum number of folders returned through the `limit` query parameter, the default is 1000. You can also pass the `page` query parameter for fetching folders from a page other than the first one.
 
+If nested folders are enabled, it expects an additional query parameter `parentUid` with the parent folder UID and returns the immediate subfolders that the authenticated user has permission to view.
+If the parameter is not supplied then it returns immediate subfolders under the root
```
Comment (imatwawana) [inline docs/sources/developers/http_api/folder.md:39]: In the first and second sentences, what is "it" replacing? The call? The operation? We should spell that out.

File: docs/sources/developers/http_api/folder.md [hunk 2]
```diff
@@ -118,6 +122,11 @@ Content-Type: application/json
 }
 ```
 
+If nested folders are enabled, and the folder is nested (lives under another folder) then the response additionally contains:
+
+- **parentUid** - the parent folder UID.
+- **parents** - an array with the whole tree hierarchy starting from the root going down up to the parent folder.
```
Comment (imatwawana) [inline docs/sources/developers/http_api/folder.md:128]: Should "down" and "up" both be in the description of **parents**? Or are we just missing a comma?
Whatever fix we make here should be applied to lines 198, 199, 269, and 270.

```suggestion
- **parentUid** - The parent folder UID.
- **parents** - An array with the whole tree hierarchy, starting from the root going down up to the parent folder.
```

File: docs/sources/developers/http_api/folder.md [hunk 3]
```diff
@@ -118,6 +122,11 @@ Content-Type: application/json
 }
 ```
 
+If nested folders are enabled, and the folder is nested (lives under another folder) then the response additionally contains:
```
Comment (imatwawana) [inline docs/sources/developers/http_api/folder.md:125]: Same thing for lines 196 and 267. 

```suggestion
If nested folders are enabled, and the folder is nested (lives under another folder), then the response additionally contains:
```

File: docs/sources/developers/http_api/folder.md [hunk 4]
```diff
@@ -329,6 +348,8 @@ Status Codes:
 
 Will return the folder identified by id.
 
+This is deprecated. Please use [get folder by UID]({{< ref "#get-folder-by-uid" >}}) instead.
```
Comment (imatwawana) [inline docs/sources/developers/http_api/folder.md:351]:
```suggestion
This is deprecated. Use [get folder by UID]({{< ref "#get-folder-by-uid" >}}) instead.
```

File: docs/sources/developers/http_api/folder.md [hunk 5]
```diff
@@ -375,3 +396,92 @@ Status Codes:
 - **401** – Unauthorized
 - **403** – Access Denied
 - **404** – Folder not found
+
+## Move folder
+
+`POST /api/folders/:uid/move`
+
+Will move the folder.
```
Comment (imatwawana) [inline docs/sources/developers/http_api/folder.md:404]:
```suggestion
Moves the folder.
```

File: docs/sources/developers/http_api/folder.md [hunk 6]
```diff
@@ -375,3 +396,92 @@ Status Codes:
...
+| Action          | Scope                                  |
+| --------------- | -------------------------------------- |
+| `folders:write` | `folders:uid:<destination folder UID>` |
+
+If moving the folder under root:
+| Action | Scope |
+| -------------- | ------------- |
+| `folders:create` | `folders:*` |
+
+JSON Body schema:
```
Comment (imatwawana) [inline docs/sources/developers/http_api/folder.md:423]:
```suggestion
JSON body schema:
```

File: docs/sources/developers/http_api/folder.md [hunk 7]
```diff
@@ -375,3 +396,92 @@ Status Codes:
...
+
+- **parentUid** – Optional [unique identifier](/http_api/folder/#identifier-id-vs-unique-identifier-uid) of the new parent folder. If it's empty then the folder is moved under the root.
```
Comment (imatwawana) [inline docs/sources/developers/http_api/folder.md:425]:
```suggestion
- **parentUid** – Optional [unique identifier](/http_api/folder/#identifier-id-vs-unique-identifier-uid) of the new parent folder. If this is empty, then the folder is moved under the root.
```

Comment (imatwawana) [inline docs/sources/developers/http_api/folder.md:425]: The link on line 425 produces a 404.

File: docs/sources/developers/http_api/folder.md [hunk 8]
```diff
@@ -375,3 +396,92 @@ Status Codes:
...
+			"updated": "2023-04-27T21:53:46.070673+03:00"
+		}
+	]
+}
+```
+
+Status Codes:
+
+- **200** – Moved
+- **400** – Errors (invalid json, missing or invalid fields, etc)
```
Comment (imatwawana) [inline docs/sources/developers/http_api/folder.md:484]:
```suggestion
- **400** – Errors (invalid JSON, missing or invalid fields, and so on)
```

File: docs/sources/developers/http_api/folder.md [hunk 9]
```diff
@@ -375,3 +396,92 @@ Status Codes:
...
+- **401** – Unauthorized
+- **403** – Access Denied
+- **404** – Not Found
```
Comment (imatwawana) [inline docs/sources/developers/http_api/folder.md:487]:
```suggestion
- **403** – Access denied
- **404** – Not found
```

**Review by imatwawana — COMMENTED** [review]
Thanks for making these updates! I've made a number of edits to conform to our style and I've left a couple questions where I think things could be clearer. Please re-request review when you're ready for me to look again.

File: docs/sources/developers/http_api/folder.md (hunk 1 above)
Comment (papagian) [reply docs/sources/developers/http_api/folder.md]: I have re-phrased it: can you check once again? thank you!

File: docs/sources/developers/http_api/folder.md [hunk 10]
```diff
@@ -35,6 +35,10 @@ that you cannot use this API for retrieving information about the General folder
 
 Returns all folders that the authenticated user has permission to view. You can control the maximum number of folders returned through the `limit` query parameter, the default is 1000. You can also pass the `page` query parameter for fetching folders from a page other than the first one.
 
+If nested folders are enabled, the operation expects an additional optional query parameter `parentUid` with the parent folder UID and returns the immediate subfolders that the authenticated user has permission to view.
```
Comment (imatwawana) [inline docs/sources/developers/http_api/folder.md:38]:
```suggestion
If nested folders are enabled, the operation expects an additional optional query parameter `parentUid` with the parent folder UID, and returns the immediate subfolders that the authenticated user has permission to view.
```

File: docs/sources/developers/http_api/folder.md [hunk 11]
```diff
@@ -35,6 +35,10 @@ that you cannot use this API for retrieving information about the General folder
...
+If the parameter is not supplied then the operation returns immediate subfolders under the root
```
Comment (imatwawana) [inline docs/sources/developers/http_api/folder.md:39]:
```suggestion
If the parameter is not supplied, then the operation returns immediate subfolders under the root
```

File: docs/sources/developers/http_api/folder.md [hunk 12]
```diff
@@ -375,3 +396,92 @@ Status Codes:
...
+| `folders:write` | `folders:uid:<destination folder UID>` |
+
+If moving the folder under root:
+| Action | Scope |
+| -------------- | ------------- |
+| `folders:create` | `folders:*` |
+
+JSON body schema:
+
+- **parentUid** – Optional [unique identifier](/http_api/folder/#identifier-id-vs-unique-identifier-uid) of the new parent folder. If this is empty, then the folder is moved under the root.
```
Comment (imatwawana) [inline docs/sources/developers/http_api/folder.md:425]:
```suggestion
- **parentUid** – Optional [unique identifier]({{< relref "#identifier-id-vs-unique-identifier-uid" >}}) of the new parent folder. If this is empty, then the folder is moved under the root.
```

**Review by imatwawana — APPROVED** [review]
I've made a couple small grammar edits and fixed the link on line 425, but otherwise, this looks good!

**Pull request merged**